streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1069,23 +996,24 @@ private StandbyTask createStandbyTask(TaskId id, Collection<TopicPartition> part
     }
 
     private void addStandbyTasks(final long start) {
-        if (partitionAssignor == null)
+        if (partitionAssignor == null) {
             throw new IllegalStateException(logPrefix + " Partition assignor has not been initialized while adding standby tasks: this should not happen.");
+        }
 
-        Map<TopicPartition, Long> checkpointedOffsets = new HashMap<>();
+        final Map<TopicPartition, Long> checkpointedOffsets = new HashMap<>();
 
         final Map<TaskId, Set<TopicPartition>> newStandbyTasks = new HashMap<>();
 
         // collect newly assigned standby tasks and reopen re-assigned standby tasks
-        for (Map.Entry<TaskId, Set<TopicPartition>> entry : partitionAssignor.standbyTasks().entrySet()) {
+        for (final Map.Entry<TaskId, Set<TopicPartition>> entry : partitionAssignor.standbyTasks().entrySet()) {
             final TaskId taskId = entry.getKey();
             final Set<TopicPartition> partitions = entry.getValue();
-            StandbyTask task = findMatchingSuspendedStandbyTask(taskId, partitions);
+            final StandbyTask task = findMatchingSuspendedStandbyTask(taskId, partitions);
 
             if (task != null) {
                 log.debug("{} recycling old standby task {}", logPrefix, taskId);
                 suspendedStandbyTasks.remove(taskId);
-                task.initTopology();
+                task.init();
```

Comment:
For standby tasks `init` is no-op, so we can just remove it? PS: logically we should call `resume` not `init` here, but the former is also no-op anyways..